core/src/stores/postgres.rs

```diff
@@ -2148,88 +2156,88 @@ impl Store for PostgresStore {
         Ok((tables, total))
     }
 
-    async fn upsert_database_row(
-        &self,
-        project: &Project,
-        data_source_id: &str,
-        database_id: &str,
-        table_id: &str,
-        row_id: &str,
-        content: &Value,
-    ) -> Result<DatabaseRow> {
-        let project_id = project.project_id();
-        let data_source_id = data_source_id.to_string();
-        let database_id = database_id.to_string();
-        let table_id = table_id.to_string();
-
-        let pool = self.pool.clone();
-        let c = pool.get().await?;
-
-        // get the data source row id
-        let stmt = c
-            .prepare(
-                "SELECT id FROM data_sources WHERE project = $1 AND data_source_id = $2 LIMIT 1",
-            )
-            .await?;
-
-        let r = c.query(&stmt, &[&project_id, &data_source_id]).await?;
-        let data_source_row_id: i64 = match r.len() {
-            0 => panic!("Unknown DataSource: {}", data_source_id),
-            1 => r[0].get(0),
-            _ => unreachable!(),
-        };
-
-        // get the database row id
-        let stmt = c
-            .prepare("SELECT id FROM databases WHERE data_source = $1 AND database_id = $2 LIMIT 1")
-            .await?;
-
-        let r = c.query(&stmt, &[&data_source_row_id, &database_id]).await?;
-
-        let database_row_id: i64 = match r.len() {
-            0 => panic!("Unknown Database: {}", database_id),
-            1 => r[0].get(0),
-            _ => unreachable!(),
-        };
-
-        // get the table row id
-        let stmt = c
-            .prepare(
-                "SELECT id FROM databases_tables WHERE database = $1 AND table_id = $2 LIMIT 1",
-            )
-            .await?;
-        let r = c.query(&stmt, &[&database_row_id, &table_id]).await?;
-        let table_row_id: i64 = match r.len() {
-            0 => panic!("Unknown Table: {}", table_id),
-            1 => r[0].get(0),
-            _ => unreachable!(),
-        };
-
-        let row_created = utils::now();
-
-        let row_id = row_id.to_string();
-        let row_data = content.to_string();
-
-        // Upsert Database Row.
-        let stmt = c
-            .prepare(
-                "INSERT INTO databases_rows \
-                   (id, database_table, created, row_id, content) \
-                   VALUES (DEFAULT, $1, $2, $3, $4, $5) \
-                   ON CONFLICT (row_id, database_table) DO UPDATE \
-                   SET content = EXCLUDED.content \
-                   RETURNING id",
-            )
-            .await?;
-
-        c.query_one(
-            &stmt,
-            &[&table_row_id, &(row_created as i64), &row_id, &row_data],
-        )
-        .await?;
-
-        Ok(DatabaseRow::new(row_created, Some(row_id), content))
-    }
+    // async fn upsert_database_row(
```

Comment:
Let's just get rid of this if we don't use it. I don't know that we ever will since we have a batch variant of this which can also work for a single row. 
If we ever miss it, we'll find it in git history.

Reply:
sgtm